File: leetcode/easy/496-next-greater/brute.cpp
/**
 * @file brute.cpp
 * @author Vortexx2
 * @brief Problem 496 - Next Greatest Element I
 * @date 10-07-2021
 * 
 * Runtime - 44 ms
 * Memory Usage - 8.2 MB
 */
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
  vector<int> nextGreaterElement(vector<int> &nums1, vector<int> &nums2)
  {
    bool found = false;
    vector<int> res(nums1.size());
    for (int i = 0; i < nums1.size(); i++)
    {
      res[i] = -1;
      found = false;

      for (int j = 0; j < nums2.size(); j++)
      {
        if (nums1[i] == nums2[j])
        {
          found = true;
        }

        if ((nums1[i] < nums2[j]) && found) {
          res[i] = nums2[j];
          break;
        }
      }
    }

    return res;
  }
};

File: leetcode/easy/496-next-greater/brute.py
"""
  @author Vortexx2
  Problem 496 - Next Greatest Number I

  Runtime - 168 ms
  Memory Usage - 14.4 MB
"""
from typing import List


class Solution:
  def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:

    for i, num in enumerate(nums1):
      found = False
      nums1[i] = -1
      for j in nums2:

        if j == num:
          found = True

        if (j > num and found):
          nums1[i] = j
          break


    return nums1
